### DISCUSSION TITLE
Accessing @error from inside a livewire component

### DISCUSSION THREAD

**Discussion by RebelSoftware** [livewire/livewire#6586] [1. Help]
I am using livewire 3

I have a form which has several embedded livewire components (they do active lookup on search phrases) the form is validated by a formRequest class from my controller. In other words validation is not done inside the Livewire component. 

So I naively added the following to my livewire components blade file 

```
                @error('registration_id') 
                <div class="alert">{{ $message }}</div>
                @enderror
```

Thinking it would work, but it doesn't, I believe this is because livewire sets up its own error handler and as the component didn't do error checking it doesn't have an error.

I have tried the following inside my mount method 
```

        if (Session::has('errors')) {
            $this->setErrorBag(Session::get('errors')->all());
        }
```

However the errors are still not displayed in the @error block

Edit: I have also tried this in my render function

```
        if (Session::has('errors')) {
            $errormessage = Session::get('errors')->first('registration_id');
            if ($errormessage) {
                $this->addError('registration_id', $errormessage);
            }
        }
```

Still the error directive appears not to be working.

What is the best way to get access to the sessions error message in my livewire component and to subsequently display them?

**Comment by RebelSoftware** [comment]
So I'll answer my own question if no-one has any input on this.

Its not a great solution but the only one I have found to work (reliably).

in my class I added a public property $error.

in my render function I added the following...
```
        if (Session::has('errors')) {
            $this->error = Session::get('errors')->first($this->inputName); // this is a parent class so the input name can change
        } else {
            $this->error = null;
        }
```

In my template I changed out the @error code with the following...

```
    @empty(!$error)
        <div class="alert">{{ $error }}</div>
    @endempty
```

It provides the functionality I needed but I feel there should be a better way using the standard laravel errors directive

**Comment by austenc** [comment]
Just ran across this too -- related to #7189

**Comment by Chrismas-tech** [comment]
I did something like this :
`<div class="input-group {{ Session::has('errors') && Session::get('errors')->has('password') ? 'is-invalid' : '' }}">`
